src/content.rs

```diff
@@ -167,6 +169,70 @@ pub fn all_pages(
     }
 }
 
+pub fn get_pages_by_glob(
+    dir: PathBuf,
+    glob: String,
+    show_unpublished: bool,
+) -> anyhow::Result<BTreeMap<String, PageValues>> {
+    let index_cache: IndexCache = pages_by_glob_load(dir, glob, show_unpublished)?;
+    Ok(index_cache.contents)
+}
+
+pub fn all_pages_load(dir: PathBuf, show_unpublished: bool) -> anyhow::Result<IndexCache> {
```

Comment:
The duplication between this and `pages_by_glob_load` is seems extensive.  Pulling out common functions, or doing something like "build the file lists then pass to a shared processing function," would ensure consistency and simplify maintenance.

Possibly even unify them into a single function which takes an optional glob expression, with None running it over all files and Some running it only over matching files.  I am not sure what is most practical.

Reply:
pulled the duplicated code into a new function for now